core/src/main/java/org/apache/carbondata/core/cache/CacheProvider.java

```diff
@@ -121,15 +141,24 @@ private void createDictionaryCacheForGivenType(CacheType cacheType, String carbo
    * @param cacheType
    */
   private void createLRULevelCacheInstance(CacheType cacheType) {
-    CarbonLRUCache carbonLRUCache = null;
-    // if cache type is dictionary cache, then same lru cache instance has to be shared
-    // between forward and reverse cache
-    if (cacheType.equals(CacheType.REVERSE_DICTIONARY) || cacheType
-        .equals(CacheType.FORWARD_DICTIONARY)) {
-      carbonLRUCache = new CarbonLRUCache(CarbonCommonConstants.CARBON_MAX_LEVEL_CACHE_SIZE,
-          CarbonCommonConstants.CARBON_MAX_LEVEL_CACHE_SIZE_DEFAULT);
-      cacheTypeToLRUCacheMap.put(CacheType.REVERSE_DICTIONARY, carbonLRUCache);
-      cacheTypeToLRUCacheMap.put(CacheType.FORWARD_DICTIONARY, carbonLRUCache);
+    boolean isDriver = Boolean.parseBoolean(CarbonProperties.getInstance()
+        .getProperty(CarbonCommonConstants.IS_DRIVER_INSTANCE, "false"));
+    if (isDriver) {
+      carbonLRUCache = new CarbonLRUCache(CarbonCommonConstants.CARBON_MAX_DRIVER_LRU_CACHE_SIZE,
+          CarbonCommonConstants.CARBON_MAX_LRU_CACHE_SIZE_DEFAULT);
+    } else {
+      // if executor cache size is not configured then driver cache conf will be used
+      String executorCacheSize = CarbonProperties.getInstance()
+          .getProperty(CarbonCommonConstants.CARBON_MAX_EXECUTOR_LRU_CACHE_SIZE);
+      if (null != executorCacheSize) {
+        carbonLRUCache = new CarbonLRUCache(CarbonCommonConstants.CARBON_MAX_DRIVER_LRU_CACHE_SIZE,
+            CarbonCommonConstants.CARBON_MAX_LRU_CACHE_SIZE_DEFAULT);
```

Comment:
correct the code....in case executor cache size use CARBON_MAX_EXECUTOR_LRU_CACHE_SIZE